csharp/ql/src/semmle/code/csharp/dataflow/flowsources/AuthCookie.qll

```diff
@@ -0,0 +1,176 @@
+/**
+ * Provides classes and predicates for detecting insecure cookies.
+ */
+
+import csharp
+import semmle.code.csharp.frameworks.microsoft.AspNetCore
+
+/**
+ * Tracks if a variable with a sensitive name is used as an argument.
+ */
+class AuthCookieNameConfiguration extends DataFlow::Configuration {
+  AuthCookieNameConfiguration() { this = "AuthCookieNameConfiguration" }
+
+  private predicate isAuthVariable(Expr expr) {
+    exists(string val |
+      (
+        val = expr.getValue() or
+        val = expr.(Access).getTarget().getName()
+      ) and
+      val.regexpMatch("(?i).*(session|login|token|user|auth|credential).*") and
+      not val.regexpMatch("(?i).*(xsrf|csrf|forgery).*")
+    )
+  }
+
+  override predicate isSource(DataFlow::Node source) { isAuthVariable(source.asExpr()) }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(Call c | sink.asExpr() = c.getAnArgument())
+  }
+}
+
+/**
+ * Tracks creation of `CookieOptions` to `IResponseCookies.Append(String, String, CookieOptions)` call as a third parameter.
+ */
+class CookieOptionsTrackingConfiguration extends DataFlow::Configuration {
+  CookieOptionsTrackingConfiguration() { this = "CookieOptionsTrackingConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) {
+    source.asExpr().(ObjectCreation).getType() instanceof MicrosoftAspNetCoreHttpCookieOptions
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(MicrosoftAspNetCoreHttpResponseCookies iResponse, MethodCall mc |
+      iResponse.getAppendMethod() = mc.getTarget() and
+      mc.getNumberOfArguments() > 2 and
```

Comment:
Shouldn't `> 2` actually be `= 3` to be more explicit?

Reply:
If a new `Append` overload was introduced, it likely would be `Append(string, string, CookieOptions, something)`. But yes, it can be equals 3 for the time being.